Mvp 39/요금 등록 플로우 리팩토링

hyeri1126 opened the pull request:
코드리뷰 요청

-완료된 작업
1. caregiver-set-price-screen -> caregiver-set-price 컴포넌트로 바꿈
2. caregiver-set-additional-price-screen -> caregiver-set-additional-price 컴포넌트로 바꿈 (caregiver-set-additional-price-screen 삭제 요청)
3. caregiver-set-price-screen에서 FlatList 사용해서 두 컴포넌트를 하나의 스크린에서 처리함
4. 이전과 다음 버튼 클릭시 페이지 이동
5. header (페이지를 나타냄) 추가


-질문
1. 요금 등록 스크린 플로우는 총 3단계로 이뤄져 있는데 3번째 스크린은 무엇인가?
2. 현재 header는 터치(페이지 이동)도 반영되지 않고 색깔?도 반영되지 않은 정적인 상태임. cg-register-step 컴포넌트를 참고하면 버튼의 상태는 총 3가지인데 컴포넌트 이동 시, 버튼이 어떤 상태이길 원하는지 정확한 요구사항을 요청함. 


질문 2번은 글로 정확한 요구사항을 설명 드리기 어려워서 연락 주시면 디스코드 음성 채널 들어가겠습니다!

hyeri1126 committed:
MVP-39 (작업중) setup

hyeri1126 committed:
MVP-39 (작업중) caregiver-set-price 컴포넌트 분리

hyeri1126 committed:
MVP-39 (작업중) caregiver-set-additional-screen 컴포넌트로 분리 후, caregiver-set-price-screen FlatList 내부에서 사용. (caregiver-set-additional-screen은 삭제해도 됨)

hyeri1126 committed:
MVP-39 (작업중) 터치에 의해 스크롤 되지 않고 버튼에 의해서만 화면 제어 기능 구현

hyeri1126 committed:
MVP-39  헤더 컴포넌트 추가

smnchoi committed:
MVP-39 헤더 기능 작동 (Header), flex: 1 삭제 (Screen)

smnchoi commented:
@hyeri1126 
# 질문 답변
1. 디스코드 채널에 질문 올려놨습니다. 확인바랍니다. https://discord.com/channels/1137734002258755654/1140140327357923419/1141692457105506407
2. https://github.com/Care-Giver/CareGiver/pull/148/commits/5862122f82d2458aba3f8bf30f972dc8cacf68c3 해당 커밋을 참조하세요.

# 피드백
1. 인풋 창 디자인이 많이 다르네요. XD 디자인 상으로는 인풋창들이 전부 아래에, 설명박스는 위로 가야 하는데 위치가 바뀐 것 같아요. 한번 확인해주시겠어요?
2. 초기에 요청드린 사안은 FlatList 를 사용하여, "저장 후 다음단계" 클릭시 좌우 스크롤이 작동하여 스크롤 애니메이션과 함께 renderItem 에 있는 컴포넌트가 좌우로 이동하는 것이었습니다. 하지만, 지금 구현해주신 코드는 스크롤 애니메이션 없이, 오히려 컴포넌트 "교체"에 가깝습니다. 이렇게 구현하시게 된 이유가 있을까요? 소통에 오해가 있던 것인지, 아니면 기존 방식으로 구현에 어려움이 있어 방식을 전환하신 것인지 궁금합니다.

smnchoi committed:
MVP-39 UI 작업완료 - 요금등록 API 호출하는 것을 제외하고, 모든 UI 작업 완료, 잘못된 useHook 코드들 수정함

smnchoi committed:
MVP-39 Merge branch 'develop' into MVP-39/요금-등록-플로우-리팩토링

# Conflicts (해결함):
#	app/components/index.ts
#	app/navigators/app-navigator.tsx
#	app/screens/cg-mypage/cg-mypage-screen.tsx

smnchoi committed:
MVP-39 레거시코드 삭제, 스크린명 변경 "caregiver-set-price-screen" ➡️ "cg-set-price-screen"

smnchoi committed:
MVP-39 "저장 후 나가기" 스크린 헤더 구현 - [주의] StackNavigatior 대신, cg-set-price-screen 스크린 컴포넌트 내부에 위치시킴
